### DISCUSSION TITLE
DeliveryGroups always come out as empty array.

### DISCUSSION THREAD

**Discussion by realpratiknikam** [Shopify/function-examples#208] [Feedback: Discounts APIs]
**Field of interest:** https://shopify.dev/docs/api/functions/reference/order-discounts/graphql/common-objects/cartdeliverygroup

**Input:**
```
query Input($customerTags: [String!] = []) {
  cart {
    lines {
      quantity
    }
    deliveryGroups {
      selectedDeliveryOption {
        code
        title
        deliveryMethodType
      }
    }
  }
  discountNode {
    metafield(<redacted>) {
      value
    }
  }
}

```

**Input response:** (when customer is on payment step of checkout)
```
{
  "cart": {
    "lines": [
      {
        "quantity": 2
      }
    ],
    "deliveryGroups": []
  },
  "discountNode": {
    "metafield": {
      "value": "<refacted>"
    }
  }
}
```

**Comment by realpratiknikam** [comment]
Related Slack threads: https://shopifypartners.slack.com/archives/C0225DTE1M1/p1676282399890539

**Comment by cgenevier** [comment]
I'm having the same issue. How do we get the associated delivery group IDs for discounting shipping? Thanks!

**Comment by morci7** [comment]
Hi, any update with this issue?

**Comment by DevAOC** [comment]
I have an issue that is similar where the discount that shows up in the cart is removed because the deliveryGroups becomes an empty array once I reach the checkout screen.. Anyone know if this is part of this issue or I am facing something else?

**Comment by jfeuchterbc** [comment]
Has there been any news on this bug?

**Reply by DevAOC** [reply]
I haven't gotten any news about this

**Comment by jfeuchterbc** [comment]
For future references and people who fall here looking for deliveryGroups in functions. Funny enough this object (deliveryGroups) does work as expected in the Checkout ui extensions where you do get the deliveryGroups even when pickup is created. Not the best solution but a workaround I did was add a Cart Property using the checkout ui when the deliveryGroups changed to or from pickup and write the info I need into the cart property. Not the best but well for now its something I guess.
